analysis_driver/dataset.py

```diff
@@ -729,6 +744,9 @@ def start(self):
             if self.get('status') != DATASET_RESUME:
                 payload['start_date'] = now()
 
+            if self.dataset.type == 'sample':
```

Comment:
I hate this so much.

Reply:
It's either this or you need to duplicate the code in `SampleDataset`.
You could also extract a `_build_start_payload` from `start` where:
```python
def _build_start_payload(self):
    payload = {
        'pipeline_used': {
            'name': self.dataset.pipeline.name,
            'toolset_type': toolset.type,
            'toolset_version': toolset.version
        }
    }
    if self.get('status') != DATASET_RESUME:
        payload['start_date'] = now()
    return payload
```
Then in `SampleDataset`
```python
def _build_start_payload(self):
    payload = super()._build_start_payload(self)
    payload['genome_used'] = self.dataset.genome_version
    return payload
````

Reply:
Could do that, but is more complex than the `if self.dataset.type == 'sample'`, which is ridiculuous but simple - will try implementing it and see how I feel. I think this is related to EdinburghGenomics/Reporting-App#112, where we store history rather than state

Reply:
Update: this is inside MostRecentProc, so we'd need to pass the payload down from SampleDataset, which is more trouble than it's worth. Current change is only 2 lines and reflects the craziness in the Reporting-App schema, so will keep it as it is.